predicate/predicate.py

```diff
@@ -62,6 +64,39 @@ def eval(self, instance):
         else:
             return ret
 
+    def add(self, data, conn_type, squash=True):
+        """
+        Adapted from `django.utils.tree.Node.add`` to handle the case of
+        repeating the same lookup.  This isn't well handled by LookupTree, so we
+        need to specially handle cases like ``P(x=1) | P(x=2)``.
+        """
+        if data in self.children:
+            return data
+        if not squash:
+            self.children.append(data)
+            return data
+        if self.connector == conn_type:
+            if (isinstance(data, Node) and not data.negated
+                    and (data.connector == conn_type or len(data) == 1)):
+                for child in data.children:
+                    self.add(child, conn_type, squash=squash)
```

Comment:
This differs from the implementation in `Node.add`, and is slightly slower than calling `list.extend`.